src/routes/index.svelte

```diff
@@ -118,51 +121,70 @@
 	const timeEnd = new Date();
 	let timeStart = new Date(timeEnd);
 	timeStart.setDate(timeStart.getDate() - 7);
-	onMount(() => {
+	onMount(async () => {
 		timeStartCall = new Date();
-		const updateProperties = chainFetches(
+		if (snapshot.some((seg) => seg.properties.name === '-')) {
+			console.log(
+				`Unknown name for segments: ${snapshot
+					.filter((seg) => seg.properties.name === '-')
+					.map((seg) => seg.properties.segment_id)}`
+			);
+		}
+		console.log('Fetching historic data');
+		await delay(1000);
+		const speedResponses = await chainFetches(
 			snapshot.map(
 				(segment) =>
 					`/api/speed-${
 						segment.properties.segment_id
 					}-${timeStart.toUTCString()}-${timeEnd.toUTCString()}`
 			),
-			1100,
-			3
-		)
-			.then((speeds) => {
-				const timeEndCall = new Date();
-				console.log(`Calls took ${timeEndCall.getTime() - timeStartCall.getTime()} ms`);
-				const updateProperties = speeds.map((speed) => aggregateTrafficSnapshotData(speed.report));
-				console.log(
-					`Unknown segments: `,
-					updateProperties.filter((p) => p.name === '').map((p) => p.segment_id)
-				);
-				const updatedSnapshot = JSON.parse(JSON.stringify(snapshot));
-				updateProperties.forEach((prop, idx) => {
-					if (prop.name !== '') {
-						updatedSnapshot[idx].properties = prop;
-					}
-				});
-				Object.values(MetricEnum)
-					.filter((m) => m.name !== MetricEnum.NONE.name)
-					.forEach((m) => {
-						const updatedMetrics = sortMetric(
-							updatedSnapshot.map((s) => s.properties.metrics?.[m.name]),
-							m.decreasing
-						);
-						updatedMetrics.forEach((um, i) => {
-							if (um) {
-								updatedSnapshot[i].properties.metrics[m.name] = um;
-							}
-						});
-					});
-				metricsAvailable = true;
-				snapshot = updatedSnapshot;
-			})
-			.catch((error) => {
-				console.log(error);
+			1000,
+			1
```

Comment:
No parallellism is possible as rate limiting occurs at the telraam API level. This then leads to the metric controls not appearing on production... There even needs to be a forced delay of 1s between consecutive calls.

This is a major drawback since that means that the loading of the metrics takes about 40 seconds. Any suggestions to improve this are welcome.